Support Compressed Whisper Format

bom-d-van opened the pull request:
Hi, this is a companying PR for supporting compressed whisper format (cwhisper) in go-whisper: https://github.com/go-graphite/go-whisper/pull/1, more details about compressed whisper format could be found in that PR.

One thing to note is that the changes in go-whisper is made to be backward compatible so there is no interruptions for existing whisper files.

This PR is set to be draft so that after the changes is accepted in go-whisper, I will update the dependency.

Looking forward to have this changes accepted and all feedback is welcome. :)

bom-d-van committed:
Add compression support

bom-d-van committed:
persister: log whisper UpdateMany error

bom-d-van committed:
Enable Compressed in store new metric creation

bom-d-van committed:
persister: add extended counter

azhiltsov commented:
It would be nice to have the example config in deploy/go-carbon.conf and README.md changes in this PR

bom-d-van committed:
Update docs for supporing compressed whisper format

bom-d-van committed:
Dep update github.com/go-graphite/go-whisper

bom-d-van committed:
Support specifying compression support in storage-schemas-conf file

This makes it possible to use both compressed and non-compressed format in the same go-carbon instance.

bom-d-van committed:
Dep update go-whisper

bom-d-van commented:
HI @lomik , I have marked this PR ready to review now. Could you take a look when you are convenient and let me know if things looks alright to you?

azhiltsov reviewed (commented):
@bom-d-van
Shouldn't we recommend to use locking by default with compressed format?

lomik commented:
I do not see any obvious problems.

bom-d-van committed:
Update docs to reminder users of enabling flock when using compressed format

bom-d-van commented on README.md:
```diff
@@ -108,7 +108,7 @@ enabled = true
 # https://github.com/lomik/go-carbon/pull/225
 hash-filenames = true
-# specify to enable/disable compressed format
+# specify to enable/disable compressed format. IMPORTANT: Only one process/thread could write to compressed whisper files at a time, especially when you are rebalancing graphite clusters (with buckytools, for example), flock needs to be enabled both in go-carbon and your tooling.
```
@azhiltsov you are right, do you think this is good enough for reminding users of this enabling flock?

azhiltsov replied:
Looks good to me, but might be additionally to that the flock config parameter need to be explicitly set to true? 
what is your take on that, @lomik ?

bom-d-van replied:
@lomik @azhiltsov I have make flock enabled by default in example and doc now.

bom-d-van committed:
Enable flock by default in doc and example config

bom-d-van commented:
@lomik @azhiltsov I think this is good enough to be merged! :D
